In [ ]:
# Load the dataset
import pandas as pd
df = pd.read_csv('C:/Users/chalani/Desktop/smartcare-ai-coursework/data/raw/smartcare_ai_dataset_1000.csv')

In [ ]:
# Check the size
df.shape

(1000, 33)

In [12]:
# Check the first 5 rows
df.head()

,record_id,patient_id,age,gender,blood_group,department,diagnosis,appointment_date,waiting_days,previous_appointments,...,consultation_fee_lkr,room_charge_lkr,lab_charge_lkr,medicine_charge_lkr,total_bill_lkr,payment_status,payment_method,no_show,readmitted_30_days,disease_risk_level
0,1,P10001,53,Male,A-,General Medicine,Migraine,2025-04-10,10,1,...,2000,0,0,11596,13596,Paid,Insurance,0,0,High
1,2,P10002,26,Male,B-,General Medicine,Diabetes,2025-05-15,2,3,...,2000,0,0,3652,5652,Paid,Insurance,0,0,Medium
2,3,P10003,22,Male,B+,Orthopedics,Back Pain,2025-07-09,22,7,...,2500,0,1200,2562,6262,Unpaid,Insurance,1,0,Medium
3,4,P10004,44,Female,AB-,Cardiology,Asthma,2025-10-16,16,1,...,2000,0,5000,10262,17262,Paid,Online,0,0,Medium
4,5,P10005,51,Female,O+,Neurology,Hypertension,2025-12-18,12,4,...,4000,0,6000,10414,20414,Paid,Cash,0,0,Medium


In [ ]:
# Check the column types
df.dtypes

record_id                         int64
patient_id                          str
age                               int64
gender                              str
blood_group                         str
department                          str
diagnosis                           str
appointment_date                    str
waiting_days                      int64
previous_appointments             int64
missed_previous_appointments      int64
appointment_status                  str
admitted                          int64
room_type                           str
length_of_stay_days               int64
previous_admissions               int64
systolic_bp                       int64
diastolic_bp                      int64
blood_sugar_mg_dl                 int64
cholesterol_mg_dl                 int64
bmi                             float64
lab_tests_count                   int64
treatments_count                  int64
consultation_fee_lkr              int64
room_charge_lkr                   int64


In [ ]:
# Load the data dictionary 
dd = pd.read_csv('C:/Users/chalani/Desktop/smartcare-ai-coursework/data/raw/smartcare_ai_dataset_data_dictionary.csv')
dd

,Column,Description
0,record_id,Unique row identifier
1,patient_id,Synthetic patient identifier
2,age,Patient age in years
3,gender,Patient gender
4,blood_group,Patient blood group
5,department,Hospital department
6,diagnosis,Primary diagnosis category
7,appointment_date,Appointment date
8,waiting_days,Number of days between booking and appointment
9,previous_appointments,Number of previous appointments


# Group columns into categories

Identifiers: record_id, patient_id

Patient info: age, gender, blood_group

Clinical: diagnosis, systolic_bp, diastolic_bp, blood_sugar_mg_dl, cholesterol_mg_dl, bmi

Operations: department, appointment_date, waiting_days, previous_appointments, 
missed_previous_appointments, appointment_status, admitted, room_type, 
length_of_stay_days, previous_admissions, lab_tests_count, treatments_count

Financial: consultation_fee_lkr, room_charge_lkr, lab_charge_lkr, medicine_charge_lkr, 
total_bill_lkr, payment_status, payment_method

Targets: no_show, readmitted_30_days, disease_risk_level

# Our target column

Our target variable is disease_risk_level (Low/Medium/High), used for Option C.
The columns no_show and readmitted_30_days are targets for Options A and B 
and are not used in this project.

In [ ]:
# Check for missing values
df.isnull().sum()

record_id                         0
patient_id                        0
age                               0
gender                            0
blood_group                       0
department                        0
diagnosis                         0
appointment_date                  0
waiting_days                      0
previous_appointments             0
missed_previous_appointments      0
appointment_status                0
admitted                          0
room_type                       906
length_of_stay_days               0
previous_admissions               0
systolic_bp                       0
diastolic_bp                      0
blood_sugar_mg_dl                 0
cholesterol_mg_dl                 0
bmi                               0
lab_tests_count                   0
treatments_count                  0
consultation_fee_lkr              0
room_charge_lkr                   0
lab_charge_lkr                    0
medicine_charge_lkr               0
total_bill_lkr              

In [14]:
# Check for duplicates
df.duplicated().sum()

np.int64(0)

In [15]:
# Investigate the missing room_type values
df[df['room_type'].isnull()]['admitted'].value_counts()

admitted
0    670
1    236
Name: count, dtype: int64

In [16]:
# Check target column balance
df['disease_risk_level'].value_counts()

disease_risk_level
Medium    469
High      400
Low       131
Name: count, dtype: int64

In [17]:
# Check for weird values
df.describe()

,record_id,age,waiting_days,previous_appointments,missed_previous_appointments,admitted,length_of_stay_days,previous_admissions,systolic_bp,diastolic_bp,...,bmi,lab_tests_count,treatments_count,consultation_fee_lkr,room_charge_lkr,lab_charge_lkr,medicine_charge_lkr,total_bill_lkr,no_show,readmitted_30_days
count,1000.000000,1000.000000,1000.000000,1000.000000,1000.000000,1000.000000,1000.000000,1000.00000,1000.000000,1000.000000,...,1000.000000,1000.000000,1000.000000,1000.000000,1000.00000,1000.000000,1000.000000,1000.000000,1000.000000,1000.000000
mean,500.500000,44.744000,21.851000,2.884000,0.553000,0.330000,1.103000,0.85500,128.423000,78.828000,...,25.519100,2.292000,1.805000,2820.500000,2412.50000,4869.600000,7971.000000,18073.600000,0.495000,0.253000
std,288.819436,17.851474,13.040892,1.693343,0.740094,0.470448,1.885191,0.95859,15.486895,10.043669,...,4.252852,1.679749,1.596723,837.694146,11140.88026,4122.515127,4629.777041,13710.258789,0.500225,0.434948
min,1.000000,1.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.00000,85.000000,50.000000,...,14.000000,0.000000,0.000000,1500.000000,0.00000,0.000000,506.000000,2954.000000,0.000000,0.000000
25%,250.750000,33.000000,11.000000,2.000000,0.000000,0.000000,0.000000,0.00000,117.000000,72.000000,...,22.700000,1.000000,1.000000,2000.000000,0.00000,2400.000000,4316.500000,11018.000000,0.000000,0.000000
50%,500.500000,44.000000,22.000000,3.000000,0.000000,0.000000,0.000000,1.00000,128.000000,79.000000,...,25.600000,2.000000,1.000000,3000.000000,0.00000,3600.000000,7680.000000,15068.500000,0.000000,0.000000
75%,750.250000,57.000000,34.000000,4.000000,1.000000,1.000000,2.000000,1.00000,139.000000,86.000000,...,28.300000,3.000000,3.000000,3500.000000,0.00000,6300.000000,11032.000000,20870.000000,1.000000,1.000000
max,1000.000000,90.000000,44.000000,10.000000,4.000000,1.000000,9.000000,5.00000,178.000000,111.000000,...,38.800000,10.000000,10.000000,4000.000000,150000.00000,27000.000000,20371.000000,179528.000000,1.000000,1.000000


## Dataset Overview
The dataset contains 1000 hospital records with 33 columns, covering patient 
demographics, clinical measurements, hospital operations, and billing data. 
There are no duplicate rows. The dataset is shared across three possible AI 
tasks (Options A, B, C) and includes three target columns — this project uses 
disease_risk_level (Option C).

## Attribute Description
Columns are grouped into 6 categories:
- Identifiers: record_id, patient_id
- Patient info: age, gender, blood_group
- Clinical: diagnosis, systolic_bp, diastolic_bp, blood_sugar_mg_dl, cholesterol_mg_dl, bmi
- Operations: department, appointment_date, waiting_days, previous_appointments, 
  missed_previous_appointments, appointment_status, admitted, room_type, 
  length_of_stay_days, previous_admissions, lab_tests_count, treatments_count
- Financial: consultation_fee_lkr, room_charge_lkr, lab_charge_lkr, 
  medicine_charge_lkr, total_bill_lkr, payment_status, payment_method
- Targets: no_show, readmitted_30_days, disease_risk_level

## Data Dictionary Interpretation
- No duplicate rows found.
- Only room_type has missing values: 906 out of 1000 rows (90.6%).
  - 670 of these are patients who were not admitted (admitted=0) — expected, 
    since room_type doesn't apply if a patient wasn't admitted.
  - 236 of these are patients who WERE admitted (admitted=1) but still have no 
    room_type recorded — this is a genuine data quality issue, not expected 
    behavior, and will be handled separately from the 670 in preprocessing.
- disease_risk_level (our target) is imbalanced: Medium 469, High 400, Low 131. 
  Low is under-represented, which will affect model training and evaluation 
  in later tasks (stratified splitting, per-class metrics needed instead of 
  plain accuracy).
- No impossible values found in numeric columns (no negative ages, charges, etc).